### PR TITLE
WIP: Use internal jpeg sources

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

xref: https://github.com/conda-forge/gdal-feedstock/issues/217

This PR attempts to revert back to the default GDAL behaviour which is to use the internal build of JPEG. This internal version appears to have diverged quite a bit from the upstream and includes support for 12 bit JPEGS (and 12 bit JPEG in TIFF). It isn't clear that upstream `jpeg` supports 12 bit and anyway GDAL doesn't allow you to have 12 bit and external `jpeg`.

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -31,7 +31,9 @@ bash configure --prefix=$PREFIX \
                --with-geotiff=$PREFIX \
                --with-hdf4=$PREFIX \
                --with-hdf5=$PREFIX \
-               --with-jpeg=$PREFIX \
+               --with-jpeg=internal \
+               --with-jpeg12 \
+               --with-hide-internal-symbols=yes \
```
Comment (gillins) [inline recipe/build.sh]: I *think* this will hide jpeg symbols so there won't be clashes with `libjpeg`. Hopefully this doesn't cause other problems....

**Comment by kmuehlbauer** [comment]
> This internal version appears to have diverged quite a bit from the upstream and includes support for 12 bit JPEGS (and 12 bit JPEG in TIFF).

@gillins Somehow 12bit JPEG in TIFF is not enabled (see [build](https://circleci.com/gh/conda-forge/gdal-feedstock/1084):
```
LIBJPEG support:           internal
12 bit JPEG:               yes
12 bit JPEG-in-TIFF:       no
```

**Comment by gillins** [comment]
There is this message from the log:
```
configure: WARNING: Internal libjpeg12 has not the same ABI as libjpeg 8 bit. Disabling JPEG-in-TIFF 12 bit
```
Seems there is something wrong with the `configure` test? (see https://github.com/OSGeo/gdal/blob/master/gdal/configure.ac#L1901). Seems like it should work.

**Comment by gillins** [comment]
Yes it does appear to be (https://github.com/conda-forge/libtiff-feedstock/blob/master/recipe/meta.yaml#L35). I'll try internal tiff. Sigh.

**Comment by kmuehlbauer** [comment]
@gillins This is beginning to get crazy. Seems we also need libgeotiff=internal now.

**Comment by mingwandroid** [comment]
Does anyone have contacts in the GDAL world who can explain why they are so keen to vendor things like this? Have they tried to upstream their changes and been rebuffed?

**Comment by kmuehlbauer** [comment]
@mingwandroid AFAIK, this is to circumvent such problems on their side.

We can ask on gdal-dev list, how we should tackle this.

**Comment by gillins** [comment]
@hobu I think you contribute to both `geotiff` and `gdal`? Do you have any thoughts on this?

**Comment by hobu** [comment]
I think the 12-bit in jpeg might be coming from https://github.com/libjpeg-turbo/libjpeg-turbo rather than regular libjpeg, btw. I don't mess with this part of the codebase at all, but I'm happy to route if there's questions.

**Comment by kmuehlbauer** [comment]
@hobu Thank you very much for this extensive insight into GDAL history and the possible path to avoid dragons out there.

**Comment by rouault** [comment]
> I think the 12-bit in jpeg might be coming from https://github.com/libjpeg-turbo/libjpeg-turbo

No, it comes for IGJ libjpeg-6b (at least that's the version of the internal libjpeg12 in GDAL). I'm not sure if this was upstreamed in libjpeg-6b (I believe so, not sure at all, as IJG libjpeg was already in a deep dormant state). libjpeg-turbo might have just imported this 12-bit support, but no idea if it is working and working with GDAL in particular.

I believe that another reason for the internal versions of the libraries was that it made it slightly more straighforward to build GDAL on Windows, as packaging of open source software on Windows has historically be a 4th-class citizen.

But otherwise using the latest upstream versions of libtiff, libgeotiff, and libjpeg with GDAL is fine.

**Comment by gillins** [comment]
Thanks @hobu and @rouault. I've tried adding `--with-jpeg12` and using the external libs but I still get the `configure: WARNING: Internal libjpeg12 has not the same ABI as libjpeg 8 bit. Disabling JPEG-in-TIFF 12 bit` message and JPEG12 in Tiff is disabled. 
`configure.ac` line 1901 seems to suggest that 12 bit JPEG in Tiff is only enabled if `JPEG_LIB_VERSION == 62` (presumeably this is the internal version). `nmake.opt` also states that "12bit libjpeg - use only with internal jpeg builds". 
[Wikipedia](https://en.wikipedia.org/wiki/Libjpeg) suggests that 12 bit was added to libjpeg in version 9a (we use 9b) but I have no idea how to test this.

I see that `libtiff` has a `jpeg12` option that we should enable, but I can't see that is going to make gdal configure happy.

I'm sure I'm missing something here - your help is very much appreciated!

**Comment by gillins** [comment]
Closing as advice appears to be to use external `jpeg` (https://github.com/conda-forge/gdal-feedstock/pull/255#issuecomment-451235642) but `gdal` currently doesn't support external `jpeg` with 12 bit (https://github.com/conda-forge/libtiff-feedstock/pull/37#issuecomment-451727931). May reopen when it does support this.

**Pull request closed without merging**